services/graph/pkg/config/config.go

```diff
@@ -45,51 +45,51 @@ type Spaces struct {
-	GroupIDAttribute   string `yaml:"group_id_attribute" env:"LDAP_GROUP_SCHEMA_ID;GRAPH_LDAP_GROUP_ID_ATTRIBUTE" desc:"LDAP Attribute to use as the unique id for groups. This should be a stable globally unique ID like a UUID."`
+	GroupSearchScope   string `yaml:"group_search_scope" env:"OCIS_LDAP_GROUP_SCOPE;LDAP_GROUP_SCOPE;GRAPH_LDAP_GROUP_SEARCH_SCOPE" desc:"LDAP search scope to use when looking up groups. Supported scopes are 'base', 'one' and 'sub'." deprecationVersion:"3.0" removalVersion:"3.1" deprecationInfo:"LDAP_GROUP_SCOPE changing name for consistency" deprecationReplacement:"OCIS_LDAP_GROUP_SCOPE"`
+	GroupFilter        string `yaml:"group_filter" env:"OCIS_LDAP_GROUP_FILTER;LDAP_GROUP_FILTER;GRAPH_LDAP_GROUP_FILTER" desc:"LDAP filter to add to the default filters for group searches." deprecationVersion:"3.0" removalVersion:"3.1" deprecationInfo:"LDAP_GROUP_FILTER changing name for consistency" deprecationReplacement:"OCIS_LDAP_GROUP_FILTER"`
+	GroupObjectClass   string `yaml:"group_objectclass" env:"OCIS_LDAP_GROUP_OBJECTCLASS;LDAP_GROUP_OBJECTCLASS;GRAPH_LDAP_GROUP_OBJECTCLASS" desc:"The object class to use for groups in the default group search filter ('groupOfNames')." deprecationVersion:"3.0" removalVersion:"3.1" deprecationInfo:"LDAP_GROUP_OBJECTCLASS changing name for consistency" deprecationReplacement:"OCIS_LDAP_GROUP_OBJECTCLASS"`
+	GroupNameAttribute string `yaml:"group_name_attribute" env:"OCIS_LDAP_GROUP_SCHEMA_GROUPNAME;LDAP_GROUP_SCHEMA_GROUPNAME;GRAPH_LDAP_GROUP_NAME_ATTRIBUTE" desc:"LDAP Attribute to use for the name of groups." deprecationVersion:"3.0" removalVersion:"3.1" deprecationInfo:"LDAP_GROUP_SCHEMA_GROUPNAME changing name for consistency" deprecationReplacement:"OCIS_LDAP_GROUP_SCHEMA_GROUPNAME"`
+	GroupIDAttribute   string `yaml:"group_id_attribute" env:"OCIS_LDAP_GROUP_SCHEMA_ID;LDAP_GROUP_SCHEMA_ID;GRAPH_LDAP_GROUP_ID_ATTRIBUTE" desc:"LDAP Attribute to use as the unique id for groups. This should be a stable globally unique ID like a UUID." deprecationVersion:"3.0" removalVersion:"3.1" deprecationInfo:"LDAP_GROUP_SCHEMA_ID changing name for consistency" deprecationReplacement:"OCIS_LDAP_GROUP_SCHEMA_ID"`
 
-	EducationResourcesEnabled bool `yaml:"education_resources_enabled" env:"LDAP_EDUCATION_RESOURCES_ENABLED;GRAPH_LDAP_EDUCATION_RESOURCES_ENABLED" desc:"Enable LDAP support for managing education related resources"`
+	EducationResourcesEnabled bool `yaml:"education_resources_enabled" env:"OCIS_LDAP_EDUCATION_RESOURCES_ENABLED;LDAP_EDUCATION_RESOURCES_ENABLED;GRAPH_LDAP_EDUCATION_RESOURCES_ENABLED" desc:"Enable LDAP support for managing education related resources" deprecationVersion:"3.0" removalVersion:"3.1" deprecationInfo:"LDAP_EDUCATION_RESOURCES_ENABLED changing name for consistency" deprecationReplacement:"OCIS_LDAP_EDUCATION_RESOURCES_ENABLED"`
```

Comment:
I'd prefer to just deprecate the global variables for the education related stuff and not reintroduce a new global setting for those. All those settings are really just specific to the graph service. So:

```
env:"GRAPH_LDAP_EDUCATION_RESOURCES_ENABLED" desc:"Enable LDAP support for managing education related resources" deprecationVersion:"3.0" removalVersion:"3.1" deprecationInfo:"LDAP_EDUCATION_RESOURCES_ENABLED removed" deprecationReplacement:"GRAPH_LDAP_EDUCATION_RESOURCES_ENABLED"`
```

The same is true for `LDAP_SCHOOL_BASE_DN`, `LDAP_SCHOOL_SEARCH_SCOPE`, `LDAP_SCHOOL_FILTER`, `LDAP_SCHOOL_OBJECTCLASS`, `LDAP_SCHOOL_SCHEMA_SCHOOL_NAME` an `LDAP_SCHOOL_SCHEMA_SCHOOL_NUMBER`.

I believe that is also what @micbar was referring to: https://github.com/owncloud/ocis/pull/6008#issuecomment-1496590386

Reply:
* If an envvar is only available in one service, it must start with the service name
* If an envvar is present in multiple services, it must start with OCIS_ as it is global anyways

If we mix things up againg we do not get out of the mess.

Reply:
all fixed, setting to resolved